src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4344,7 +4344,7 @@ private BoundExpression MakeBadExpressionForObjectCreation(ObjectCreationExpress
                     typeArgumentsSyntax: default(SeparatedSyntaxList<TypeSyntax>),
                     typeArgumentsWithAnnotations: default(ImmutableArray<TypeWithAnnotations>),
                     invoked: false,
-                    indexed: false,
+                    indexed: allInitialiserExpressionsAreIndexExpression(),
```

Comment:
It feels like it would be better to just make this a local above this statement.